internal/deploy/bootstrap.go

```diff
@@ -63,16 +73,59 @@ func Bootstrap(
 		manifests = append(manifests, []byte(wildCardRoleAndAssignment)...)
 	}
 
-	if err := applyManifests(
+	manifestObjs, err := applyManifests(
 		ctx, k8s, manifests, applyOpts{
 			dryRun, force, defaultRetries, defaultInitialBackoff},
-	); err != nil {
+	)
+	if err != nil {
+		return false, err
+	}
+
+	if _, err = PatchDeploymentWithInKcpModeFlag(ctx, k8s, manifestObjs, isInKcpMode); err != nil {
 		return false, err
 	}
 
 	return hasKyma(string(manifests))
 }
 
+func PatchDeploymentWithInKcpModeFlag(ctx context.Context, k8s kube.KymaKube, manifestObjs []ctrlClient.Object, isInKcpMode bool) (*appsv1.Deployment, error) {
+	var patchedDeployment *appsv1.Deployment
+	for _, manifest := range manifestObjs {
+		if manifest.GetObjectKind().GroupVersionKind().Kind == "Deployment" {
+			hasKcpFlag := false
+			if manifestObj, success := manifest.(*unstructured.Unstructured); success {
+				manifestJSON, err := json.Marshal(manifestObj.Object)
+				if err == nil {
+					deploymentSpec := appsv1.Deployment{}
+					if err = json.Unmarshal(manifestJSON, &deploymentSpec); err == nil {
```

Comment:
This level of nesting is pretty crazy, please just extract some function for the deployment lookup where you used early returns